standards/api_security_standards.md

```diff
@@ -0,0 +1,158 @@
+### **INTRODUCTION**
+
+This standard is developed to guide secure API implementation across our digital infrastructure, aligning with the OWASP Top 10 security risks for APIs. It outlines mandatory practices that help protect against common vulnerabilities and ensure consistent application of security controls.
+
+### **SCOPE**
+
+This standard applies to all IT systems and applications developed either internally or for a client by Symph Inc.
+
+### **STANDARDS**
+
+**1. Broken Object Level Authorization**
+
+APIs tend to expose endpoints that handle object identifiers, creating a wide attack surface Level Access Control issue. Object level authorization checks should be considered in every function that accesses a data source using an input from the user.
+
+1.1 Rigorous Authorization Checks: Ensure every API endpoint that receives an object ID implements robust object level authorization checks. These checks must confirm the user's rights to access or manipulate the specified object.
+
+1.2 User Policies and Hierarchy Enforcement: Utilize user policies and hierarchical roles to govern access rights, ensuring that authorization logic is consistently applied across all API interactions.
+
+1.3 Secure Object Identifiers: Use random and unpredictable values, such as GUIDs, for object identifiers to prevent attackers from guessing and manipulating object IDs.
+
+1.4 Testing and Verification: Regularly test the authorization mechanisms to identify and rectify any flaws. Ensure that changes that could introduce vulnerabilities are not deployed without thorough testing.
+
+**2. Broken Authentication**
+
+Authentication mechanisms are often implemented incorrectly, allowing attackers to compromise authentication tokens or to exploit implementation flaws to assume other user's identities temporarily or permanently. Compromising system's ability to identify the client/user, compromises API security overall.
+
+2.1 Robust Authentication Flows: Ensure all authentication flows are thoroughly designed and tested, including mobile, web, and any deep links that implement one-click authentication.
+
+2.2 Endpoint Security: Treat all authentication endpoints, including credential recovery and password reset, with high-security measures such as rate limiting, captcha, and account lockout mechanisms.
+
+2.3 Strong Authentication Practices: Utilize industry-standard methods for authentication, token generation, and password storage. Avoid reinventing these mechanisms.
```

Comment:
Let's add examples for these industry-standard methods and links as well if applicable

Reply:
Done adding samples